Documentation/Open Location Code API Specification.md

```diff
@@ -0,0 +1,78 @@
+
+## OPTIONAL public methods
+
+An implementation MAY implement these methods.
+
+| Method name        | Input(s)                                                     | Output(s)                                                    |
+| ------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| `isValid`          | Plus Code string                                             | True if this is a Plus Code (full code or short code), false otherwise |
+| `isShort`          | Plus Code string                                             | True if this is a short code Plus Code, false otherwise      |
+| `encode`           | Latitude, longitude, code length                             | The full code Plus Code with specified code length with area that contains the specified coordinates |
+| `decode`           | Full code Plus Code string                                   | The southern parallel and western meridian for the Plus Code area |
+| `shorten`          | Full code Plus Code, reference location latitude and longitude | The short code Plus Code representing the input where all significant digits that are allowed to be omitted are omitted; or if no digits are allowed to be omitted then the input is returned |
+| `recoverNearest`   | Short code Plus Code, reference location latitude and longitude | The unique full code Plus Code nearest (latitudinal distance + longitudinal distance) to the reference location |
+
+Implementations are RECOMMENDED to use the method names above. Note that the OPTIONAL `decode` public method overrides and tightens the specification for the REQUIRED `decode` public method. Note that The OPTIONAL `encode` public method adds a parameter. This MAY be implemented as an overloaded method or another way that is customary for the programming language.
+
+An implementation MAY implement `shorten` using separate `shortenBy4` and `shortenBy6` methods.
+
+Note: it is possible to implement `isValid` and `isShort` using a Perl Compatible Regular Expression, see the Plus Code specification.
+
+## REQUIREMENTS for all public methods
+
+Implementations MUST follow these REQUIREMENTS for all public methods:
+
+* Plus Code Inputs MUST treat non-upper-case inputs as if they were upper case.
+* Plus Code outputs MUST be uppercase and conform to the [Plus Codes Specification].
+* Latitude inputs MUST treat a latitude greater than 90° N as if it were the North Pole (i.e. at 90° N).
+* Latitude inputs MUST treat a latitude lower than 90° S as if it were the South Pole (i.e. at 90° S).
+* Longitude inputs MUST treat values outside the range from (including) 180° W to (excluding) 180° E as if they were the equivalent longitude in this range.
+* Longitude inputs MUST NOT cause runtime performance linearly dependent on the longitude. E.g. use `LONGITUDE % 360`, not `WHILE (LONGITUDE > 180) LONGITUDE -= 360`.
```

Comment:
Already added as a PR to yours, but now in review form: `LONGITUDE % 360` should be replaced by `MOD(LONGITUDE,360)` to avoid faulty implementations in languages where `%` is not the module but the remainder operator (Java being one such case).

Reply:
Fixed at 89063bda8c46c2a4964cc8d3b5b56c0e2a5b557e